exercises/practice/grains/grains.odin

```diff
@@ -0,0 +1,18 @@
+package grains
+
+Error :: enum {
+	None = 0
+	// Implement me!
+}
+
+// Returns the number of grains on the specified square.
+square :: proc(n: int) -> (u64, Error) {
+	// Implement me!
+	return 0, .None
```

Comment:
Possible to raise an exception telling the student to implement this code? That'll let us be consistent across exercises that might return other types as well.

Reply:
I've added an error condition for this exercise. Odin [will never have exceptions](https://www.gingerbill.org/article/2018/09/05/exceptions-and-why-odin-will-never-have-them/), so the pattern is either to return an error type along with the expected value, or to have have a value that will always fail the test (e.g. -1). For a case like this one (where actual invalid inputs exist) it makes sense to return an error result. In other procedures (e.g. to calculate a simple mathematical operation on any input) an error result may be unnecessary.

Reply:
Makes sense.

Reply:
As per @ErikSchierboom's suggestion, I rather changed this into a compile-time panic, which is much clearer and does not require the student to mess around with error types unnecessarily.